Fixed a possible segfault during garbage collection of ``_asyncio.FutureIter`` objects
